packages/loopring_v3/contracts/amm/libamm/AmmTransactionReceiver.sol

```diff
@@ -94,19 +101,110 @@ library AmmTransactionReceiver
             signature.offset := add(signature.offset, 0x20)
         }
         if (txType == AmmData.PoolTxType.JOIN) {
+            ctx.approveAmmUpdates(true);
             S.processJoin(
                 ctx,
                 abi.decode(data, (AmmData.PoolJoin)),
                 signature
             );
+            ctx.approveAmmUpdates(false);
         } else if (txType == AmmData.PoolTxType.EXIT) {
+            ctx.approveAmmUpdates(true);
             S.processExit(
                 ctx,
                 abi.decode(data, (AmmData.PoolExit)),
                 signature
             );
+            ctx.approveAmmUpdates(false);
```

Comment:
Why wrapping this tx but not deposit/withdrawal?

Reply:
We don't need balances/virtual balance L2 data on L1 to be able to correctly process deposits/withdrawals. Virtual balances also don't need to get updated just by moving funds between L1 and L2.